### PR TITLE
Added upload page and file list UI

### PR DESCRIPTION
**Pull request by dnarcese**

### CODE REVIEW

**Comment by dannycoates** [comment]
@dnarcese again, sorry about the conflicts in `upload.js`. I think in order to resolve it we should be able to just replace the dom elements in the new `frontend/src/upload.js` with yours. I tried it but haven't tested it here: https://gist.github.com/dannycoates/36c953449f641897f50163a30404ada7

File: .gitignore [hunk 1]
```diff
@@ -1,4 +1,11 @@
 .DS_Store
 node_modules
+<<<<<<< HEAD
+<<<<<<< HEAD
+public/bundle.js
+=======
+>>>>>>> add gitignore
+=======
+>>>>>>> d7e40776401060de6d84771a442ddb6f3ea7175d
```
Comment (pdehaan) [inline .gitignore]: I think there may have been a merge conflict here.

File: frontend/src/upload.js [hunk 2]
```diff
@@ -3,18 +3,21 @@ const FileSender = require('./fileSender');
...
   const file = event.target.files[0];
 
-  let li = document.createElement('li');
-  let name = document.createElement('p');
-  name.innerText = file.name;
-  li.appendChild(name);
+  var row = document.createElement("tr");
+  var name = document.createElement("td");
+  var link = document.createElement("td");
+  var expiry = document.createElement("td");
```
Comment (ericawright) [inline frontend/src/upload.js]: try to use "let" and "const" rather than "var"

File: frontend/src/upload.js [hunk 3]
```diff
@@ -23,21 +26,23 @@ let onChange = event => {
...
+    var btn = document.createElement("button");
+    btn.innerHTML = "x";
+    btn.style = "padding: 0; border: none; background: none; cursor: pointer"
     btn.addEventListener('click', () => {
       FileSender.delete(
         info.fileId,
         localStorage.getItem(info.fileId)
       ).then(() => {
-        document.getElementById('uploaded_files').removeChild(li);
+        document.getElementById('uploaded-files').removeChild(row);
```
Comment (ericawright) [inline frontend/src/upload.js]: you're searching for the  'uploaded-files' element more than once, do you want to make it a variable?

File: public/index.html [hunk 4]
```diff
@@ -3,16 +3,46 @@
 <head>
   <title>Firefox Fileshare</title>
   <script src="bundle.js"></script>
-
+  <link rel="stylesheet" href="https://code.cdn.mozilla.net/fonts/fira.css">
+  <link rel="stylesheet" type="text/css" href="main.css" />
 </head>
-<body>
+<body background="resources/background.png">
```
Comment (ericawright) [inline public/index.html]: is there a reason to inline this rather than style it from the css file?

File: public/upload.js [hunk 5]
```diff
@@ -0,0 +1,147 @@
+function onChange(event) {
```
Comment (ericawright) [inline public/upload.js]: there's another file `frontend/src/upload.js` which I believe this is based off of. they should probably be combined.

File: public/upload.js [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+    return function updateProgress(e) {
+      if (e.lengthComputable) {
+        var percentComplete = Math.floor((e.loaded / e.total) * 100);
+        a_element.innerHTML = "Progress: " + percentComplete + "%";
+
+        if (percentComplete === 100) {
+          var del = document.createElement("td");
+          var btn = document.createElement("button");
+          btn.innerHTML = "x";
+          btn.style = "padding: 0; border: none; background: none; cursor: pointer"
```
Comment (ericawright) [inline public/upload.js]: You can add a class to the button when the status changes, then style that class from the css file.

File: public/index.html [hunk 7]
```diff
@@ -3,16 +3,46 @@
...
-  </form>
+  <div class="main-window">
+    <div id="page-one">
+      <div class="title">
+        Share your files quickly, privately and securely.
+      </div>
+      <div class="upload-window">
+        <div id="upload-img"><img src='resources/upload.svg'/></div>
+        <div>
+          DRAG & DROP
```
Comment (ericawright) [inline public/index.html]: &amp; instead of &

File: public/index.html [hunk 8]
```diff
@@ -3,16 +3,46 @@
...
-  <form method="post" action="upload" enctype="multipart/form-data">
-    <input type="file" onchange="onChange(event)" name="fileUploaded" />
-  </form>
+  <div class="main-window">
+    <div id="page-one">
+      <div class="title">
+        Share your files quickly, privately and securely.
+      </div>
+      <div class="upload-window">
+        <div id="upload-img"><img src='resources/upload.svg'/></div>
```
Comment (ericawright) [inline public/index.html]: double " for html please

**Pull request merged**